### ISSUE TITLE
中文支持吗

### ISSUE THREAD

**Issue by gujiacheng** [beichenzbc/Long-CLIP#58]

**Comment by beichenzbc** [comment]
不支持，主要的原因是缺乏中文长文本图文数据集

**Comment by gg22mm** [comment]
可以考虑这种方案魔改再训练，提供的思路：

1、冻结Long-CLIP 只用来获取图像编码
2、训练文本编码器（我们使用chinese-roberta-wwm作为语言的编码器来支持中文）
通过这两步后，训练出来自己的clip,就可以支持中文了

最后： 用自己训练出来的clip做中文的编码器，然后用Long-CLIP 默认的做为图像编码器  ，这样就是一个玩整的，支持中文的Long-CLIP了 ~~

------------参考例子-------
'''
https://hf-mirror.com/IDEA-CCNL/Taiyi-CLIP-Roberta-102M-Chinese
首个开源的中文CLIP模型，1.23亿图文对上进行预训练的文本端RoBERTa-base。

采用兼容模式来训练：
	以获得强大的视觉-语言表征。在训练中文版的CLIP时
	我们使用chinese-roberta-wwm作为语言的编码器:https://hf-mirror.com/hfl/chinese-roberta-wwm-ext-large
	并将CLIP中的ViT-B-32应用于视觉的编码器		 :https://openaipublic.azureedge.net/clip/models/40d365715913c9da98579312b702a82c18be219cc2a73407c4526f58eba950af/ViT-B-32.pt 《重点：这里换成成：Long-CLIP》
	为了快速且稳定地进行预训练，我们冻结了视觉编码器并且只微调语言编码器。
	此外，我们将Noah-Wukong数据集(100M)和Zero数据集(23M)用作预训练的数据集，训练了24个epoch，在在A100x32上训练了7天。据我们所知，我们的Taiyi-CLIP是目前Huggingface社区中首个的开源中文CLIP。
'''

#方式四、

#文字特征 - 来自封神clip （采用兼容openai的模式来训练） 《重点：这里换成成：自己训练出来的模型》
model = BertForSequenceClassification.from_pretrained("IDEA-CCNL/Taiyi-CLIP-Roberta-102M-Chinese").eval()
tokenizer = BertTokenizer.from_pretrained("IDEA-CCNL/Taiyi-CLIP-Roberta-102M-Chinese")
query_texts = ["一只猫", "一只狗",'两只猫', '两只老虎','一只老虎']  # 这里是输入文本的，可以随意替换。
text_to_token = tokenizer(query_texts, return_tensors='pt', padding=True)['input_ids']
text_features = model(text_to_token).logits
print('---------------5---text_features.last_hidden_state:',text_features.shape,text_features) # ---------------------torch.Size([5, 512])

#图片特征 《重点：这里换成成：Long-CLIP》
url = "http://images.cocodataset.org/val2017/000000039769.jpg"  #这里可以换成任意图片的url
processor = CLIPProcessor.from_pretrained("openai/clip-vit-base-patch32")
model = CLIPModel.from_pretrained("openai/clip-vit-base-patch32").eval() 
image = processor(images=Image.open(requests.get(url, stream=True).raw), return_tensors="pt") #图片格式化标准输入


#加载CLIP的image encoder model

image_features = model.get_image_features(**image)   #图片特征 - 这人可以存到es中 叫：Knn相似度查询
print('---------------6---图片_features.last_hidden_state:',image_features.shape) #------------------------------------torch.Size([1, 512])
